cpp/src/rolling/grouped_rolling.cu

```diff
@@ -205,57 +207,62 @@ std::unique_ptr<column> grouped_rolling_window(table_view const& group_keys,
 
 namespace {
 
-bool is_supported_range_frame_unit(cudf::data_type const& data_type)
+template <typename T, std::enable_if_t<std::numeric_limits<T>::is_signed>* = nullptr>
+__device__ T add_and_check_overflow(T const& value, T const& delta)
 {
-  auto id = data_type.id();
-  return id == cudf::type_id::TIMESTAMP_DAYS || id == cudf::type_id::TIMESTAMP_SECONDS ||
-         id == cudf::type_id::TIMESTAMP_MILLISECONDS ||
-         id == cudf::type_id::TIMESTAMP_MICROSECONDS || id == cudf::type_id::TIMESTAMP_NANOSECONDS;
+  // delta >= 0.
+  return (value < 0 || (std::numeric_limits<T>::max() - value) >= delta)
+           ? (value + delta)
+           : std::numeric_limits<T>::max();
 }
 
-/// Fetches multiplication factor to normalize window sizes, depending on the datatype of the
-/// timestamp column. Used for time-based rolling-window operations. E.g. If the timestamp column is
-/// in TIMESTAMP_SECONDS, and the window sizes are specified in DAYS, the window size needs to be
-/// multiplied by `24*60*60`, before comparisons with the timestamps.
-size_t multiplication_factor(cudf::data_type const& data_type)
+template <typename T, std::enable_if_t<!std::numeric_limits<T>::is_signed>* = nullptr>
+__device__ T add_and_check_overflow(T const& value, T const& delta)
```

Comment:
Maybe `add_safe`? I've seen similarly named functions in the code base, maybe the meaning more more specific than I understood.